src/components/KDateTimePicker/CalendarWrapper.vue

```diff
@@ -62,20 +83,31 @@ import { DatePicker } from 'v-calendar'
 import type { DatePickerModel, DatePickerRangeObject, DateTimePickerMode } from '@/types'
 import { format } from 'date-fns'
 
-const props = defineProps<{
+const props = withDefaults(defineProps<{
   isRange: boolean
   kDatePickerMode: DateTimePickerMode
   maxDate?: Date
   minDate?: Date
-}>()
+  errorMessage?: string
+}>(), {
+  maxDate: undefined,
+  minDate: undefined,
+  errorMessage: 'Start time cannot exceed end time.',
+})
 const calendarVModel = defineModel<DatePickerModel>({ required: true })
+const hasError = defineModel<boolean>('error', { default: false })
 const startTimeValue = ref<string>(format(new Date(), 'HH:mm:ss'))
 const endTimeValue = ref<string>(format(new Date(), 'HH:mm:ss'))
+const componentId = ref<string>(crypto.randomUUID())
```

Comment:
issue: you don't need this to be a ref, and don't need to use crypto; Vue has a built-in for this
```suggestion
const componentId = useId()
```

Reply:
done